.github/workflows/e2e.yml

```diff
@@ -27,41 +31,65 @@ jobs:
         with:
           fetch-depth: 0
 
-      - name: Install devbox
-        uses: jetify-com/devbox-install-action@v0.11.0
+      - uses: cachix/install-nix-action@V27
+        if: inputs.provider == 'Nutanix'
         with:
-          enable-cache: true
+          github_access_token: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Go cache
-        uses: actions/cache@v4
+      - name: Install devbox
+        uses: jetify-com/devbox-install-action@v0.11.0
         with:
-          path: |
-            ~/.cache/go-build
-            ~/go/pkg/mod
-          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
-          restore-keys: |
-            ${{ runner.os }}-go-
+          enable-cache: "true"
+          skip-nix-installation: ${{ inputs.provider == 'Nutanix' }}
 
       # The default disk size of these runners is ~14GB, this is not enough to run the e2e tests.
       # Cleanup the disk, see upstream discussion https://github.com/actions/runner-images/issues/2840.
       - name: Cleanup Disk Space
+        if: inputs.runs-on != 'self-hosted-ncn-dind'
```

Comment:
Why make this conditional? Just clean it up anyway, it doesn't take any time, especially compared to the other steps.

Reply:
because the reason for the cleanup is lack of disk space in hosted  runners which is not something we're constrained by here.